src/common/variables.c

```diff
@@ -285,49 +285,44 @@ static char *_variables_get_latitude(dt_variables_params_t *params)
   }
 }
 
-static int _get_parameters(char **variable, char **parameters, size_t max_param)
-/*
+/**
  * @param char **variable - The variable to read parameters from
- * @param char **parameters - a pointer to where list of parameters are stored in
+ * @param GList **params - a pointer to a GList where the parameters are stored in
  * @param size_t max_param - The maximum number of parameters to read
- *                           (and parameters has space for pointers)
  * @return The number of parameters, -1 if any error
- *
- * You should free the string in parameters with g_free() after usage unless
- * returnvalue is 0 or -1;
  */
+static int _get_parameters(char **variable, GList **params, size_t max_param)
 {
-  *parameters = NULL;
-  if(*variable[0] == '[')
+  if(*variable[0] != '[') return -1;
+
+  (*variable)++;
+  if(*variable[0] == ',') return -1;
+
+  gchar *parameters = g_strdup(*variable);
+  char *end = g_strstr_len(parameters, -1, "]");
+  if(end)
   {
-    (*variable) ++;
-    if(*variable[0] == ',')
+    end[0] = '\0';
+    (*variable) += strlen(parameters) + 1;
+
+    size_t count = 0;
+    char *token = strtok(parameters, ",");
```

Comment:
`strtok` is not thread-safe (don't know if that's a problem here). There's `strtok_r` but it looks like you're just doing a `g_strsplit` here?

Reply:
I didn't touch that part of the code, was already there.
Will take a deeper look though.

Reply:
I haven't looked closely either (and don't want to!) Is this a linked list with only ever one or two entries? I'm not sure at all what's going on here. Why not just two pointer variables/parameters, the second of which might be NULL? Then `_get_parameters` doesn't need to be passed max_param (if the second pointer-to-pointer is NULL then you only want 1 reply) and it doesn't need to return the number found (if the second pointer is returned as NULL, you didn't find more than 1 and if the first is NULL too you didn't find any). Anyway, I'll let you have fun with this!